qiskit/transpiler/passes/optimization/template_matching/template_substitution.py

```diff
@@ -213,6 +222,12 @@ def _remove_impossible(self):
         list_predecessors = []
         remove_list = []
 
+        # First remove any scenarios that have parameters in the template.
+        # Check that all parameters have been bound, return None if there are free parameters
```

Comment:
comment says `return None`, but I don't see where this is happening

Reply:
That's obviously wrong. Fixed in a9b9737.